### PR TITLE
CB-246 Do not use WebElement.clear() in WebDriverElement.sendString()

### PR DESCRIPTION
**Pull request by dsafar**
Also several test fixes.

Green system tests: http://ci.sugarcrm.net/job/CandyBean/job/cb-dev-sys-linux/2337/
Green unit tests: http://ci.sugarcrm.net/job/CandyBean/job/cb-dev-unit-linux-sonar/1997/

### CODE REVIEW

File: src/main/java/com/sugarcrm/candybean/automation/webdriver/WebDriverElement.java [hunk 1]
```diff
@@ -353,48 +352,42 @@ public void scroll(Location loc) throws CandybeanException {
...
 	/**
 	 * Send a string to this element.
 	 * 
-	 * @param input
-	 *            string to send
-	 * @param append
-	 *            if append is true, the element will be cleared first
-	 * @throws CandybeanException
+	 * @param   input   string to send
+     * @param   append  if append is false, the element will be cleared first
```
Comment (jmittert) [inline src/main/java/com/sugarcrm/candybean/automation/webdriver/WebDriverElement.java]: This line is indented with spaces while the lines around it are indented with tabs.

File: resources/html/test/elements.html [hunk 2]
```diff
@@ -0,0 +1,24 @@
+<!--
+This file presents an html page with a single element
+that blinks every 5 seconds once clicked. It is
```
Comment (jmittert) [inline resources/html/test/elements.html]: This comment looks like it's not supposed to be here.

File: src/main/java/com/sugarcrm/candybean/automation/webdriver/WebDriverElement.java [hunk 3]
```diff
@@ -353,48 +352,42 @@ public void scroll(Location loc) throws CandybeanException {
...
-			// Re-find the element to avoid the stale element problem.
-			this.we = this.wd.findElements(Hook.getBy(this.hook)).get(this.index);
-			this.we.sendKeys(input);
-		}
+	public void sendString(String input, boolean append) throws CandybeanException {
+		logger.info((append ? "Clearing field and s" : "S") + "ending string: " + input +
+                " to element: " + toString());
+
+		if(!append) {
+            selectAll();
```
Comment (jmittert) [inline src/main/java/com/sugarcrm/candybean/automation/webdriver/WebDriverElement.java]: Here too, the whitespace mixes tabs and spaces.

File: src/test/java/com/sugarcrm/candybean/automation/webdriver/WebDriverElementSystemTest.java [hunk 4]
```diff
@@ -317,21 +321,23 @@ public void checkBoxSelectTest() throws Exception {
...
-		iface.getWebDriverElement(Strategy.ID, "search_form_input_homepage").sendString(searchString);
-		iface.getWebDriverElement(Strategy.ID, "search_button_homepage").click();
-		Assert.assertTrue(iface.getWebDriverElement(Strategy.PLINK, "SugarCRM").isDisplayed());
+        iface.go(elementsPage);
+
+        WebDriverElement textField = iface.getWebDriverElement(Strategy.ID, "textfield");
+
+		// clear and set scenario
+        textField.sendString(garbageString);
+        textField.sendString(searchString + Keys.TAB);
```
Comment (jmittert) [inline src/test/java/com/sugarcrm/candybean/automation/webdriver/WebDriverElementSystemTest.java]: Do we need to send the tab key here? I looks like all it does it focus the dropdown. Are we testing for something besides the strings matching?

**Comment by jmittert** [comment]
Tests look good on my local machine. Fix the whitespace and comment issues and it should be good to merge.

**Comment by dsafar** [comment]
Thanks, Jason.  Updated; I will post test results when they're ready.

**Comment by dsafar** [comment]
Green unit tests: http://ci.sugarcrm.net/job/CandyBean/job/cb-dev-unit-linux-sonar/2001/
Green system tests: http://ci.sugarcrm.net/job/CandyBean/job/cb-dev-sys-linux/2342/

**Comment by jmittert** [comment]
There are still a few whitespace inconstancies in sendStringTest, but otherwise looks good to merge to me.

**Comment by dsafar** [comment]
Turned on visible whitespace and fixed indentation in sendStringTest (and some that I missed in checkBoxSelectTest).

**Comment by dsafar** [comment]
Green unit tests: http://ci.sugarcrm.net/view/Folders/job/CandyBean/job/cb-dev-unit-linux-sonar/2003/
Green system tests: http://ci.sugarcrm.net/view/Folders/job/CandyBean/job/cb-dev-sys-linux/2345/

**Pull request merged**